### PR TITLE
feat: add Agentless support in generate command

### PR DESCRIPTION
**Pull request by PengyuanZhao**
## Summary

- Update the generate command to support [GCP Agentless](https://github.com/lacework/terraform-gcp-agentless-scanning) integration.
- Add the following flags: 
  - `--agentless`
  - `--regions`
  - `--project_filter_list`
 
## How did you test this change?

https://github.com/lacework/go-sdk/assets/7945666/d8d8c12a-254a-4045-b7c1-34635154f9d5

Generated terraform code:
```
terraform {
  required_providers {
    lacework = {
      source  = "lacework/lacework"
      version = "~> 1.0"
    }
  }
}

provider "google" {
  alias   = "us-east1"
  project = "preflight-project-pengyuan"
  region  = "us-east1"
}

module "lacework_gcp_agentless_scanning_global" {
  source              = "lacework/agentless-scanning/gcp"
  version             = "~> 0.1"
  global              = true
  integration_type    = "ORGANIZATION"
  organization_id     = "121868925203"
  project_filter_list = ["agentless-testing"]
  regional            = true

  providers = {
    google = google.us-east1
  }
}

module "gcp_organization_level_config" {
  source          = "lacework/config/gcp"
  version         = "~> 2.3"
  org_integration = true
  organization_id = "121868925203"

  providers = {
    google = google.us-east1
  }
}

module "gcp_organization_level_audit_log" {
  source                       = "lacework/audit-log/gcp"
  version                      = "~> 3.4"
  org_integration              = true
  organization_id              = "121868925203"
  service_account_name         = module.gcp_organization_level_config.service_account_name
  service_account_private_key  = module.gcp_organization_level_config.service_account_private_key
  use_existing_service_account = true

  providers = {
    google = google.us-east1
  }
}
```

## Issue
https://lacework.atlassian.net/browse/GROW-2668

### CODE REVIEW

File: lwgenerate/gcp/gcp.go [hunk 1]
```diff
@@ -448,26 +482,116 @@ func createLaceworkProvider(laceworkProfile string) (*hclwrite.Block, error) {
...
-	).ToBlock()
-	if err != nil {
-		return nil, err
+	return blocks, nil
+}
+
+func createAgentless(args *GenerateGcpTfConfigurationArgs) ([]*hclwrite.Block, error) {
+	if !args.Agentless {
+		return nil, nil
+
```
Comment (ipcrm) [inline lwgenerate/gcp/gcp.go:530]:
```suggestion
```

File: lwgenerate/gcp/gcp.go [hunk 2]
```diff
@@ -448,26 +482,116 @@ func createLaceworkProvider(laceworkProfile string) (*hclwrite.Block, error) {
...
+	}
+
+	blocks := []*hclwrite.Block{}
+
+	regions := append([]string{}, args.Regions...)
+	if len(regions) == 0 {
+		regions = append(regions, "")
 	}
```
Comment (ipcrm) [inline lwgenerate/gcp/gcp.go:538]: Is a valid region required? Should this be an error if it's empty?

File: lwgenerate/gcp/gcp.go [hunk 3]
```diff
@@ -448,26 +482,116 @@ func createLaceworkProvider(laceworkProfile string) (*hclwrite.Block, error) {
...
+		if i == 0 {
+			attributes["global"] = true
+			if len(args.ProjectFilterList) > 0 {
+				projectIDs := []string{}
+				for _, id := range args.ProjectFilterList {
+					projectIDs = append(projectIDs, fmt.Sprintf("\"%s\"", id))
+				}
+				attributes["project_filter_list"] = lwgenerate.CreateSimpleTraversal([]string{
+					fmt.Sprintf("[%s]", strings.Join(projectIDs, ", ")),
+				})
```
Comment (ipcrm) [inline lwgenerate/gcp/gcp.go:556]: nit: can this just be an `[]string`?

File: lwgenerate/gcp/gcp.go [hunk 4]
```diff
@@ -448,26 +482,116 @@ func createLaceworkProvider(laceworkProfile string) (*hclwrite.Block, error) {
...
+			}
+		}
+		if i > 0 {
+			moduleName = "lacework_gcp_agentless_scanning_region_" + region
+			attributes["global_module_reference"] = lwgenerate.CreateSimpleTraversal(
+				[]string{"module", "lacework_gcp_agentless_scanning_global"},
+			)
+		}
+
+		if region != "" {
```
Comment (ipcrm) [inline lwgenerate/gcp/gcp.go:570]: Similar question to [L537](https://github.com/lacework/go-sdk/pull/1500/files#diff-9406e9de58811734965d500cfe472be2bc13f42a986ddf34ce16f8817ec298feR537); why would we want an empty region value? Should that be an error?

File: lwgenerate/gcp/gcp.go [hunk 5]
```diff
@@ -534,6 +658,17 @@ func createConfiguration(args *GenerateGcpTfConfigurationArgs) ([]*hclwrite.Bloc
...
+		// Regions is required when Agentless integration is enabled
+		// Use the first region for Google provider if multiple regions are provided
+		if len(args.Regions) > 0 {
+			moduleDetails = append(
+				moduleDetails,
+				lwgenerate.HclModuleWithProviderDetails(
+					map[string]string{"google": fmt.Sprintf("google.%s", args.Regions[0])},
+				),
+			)
+		}
```
Comment (ipcrm) [inline lwgenerate/gcp/gcp.go:670]: Does the configuration integration need to be modified for agentless to work properly? If so, do we need to gate this block from running if agentless isn't enabled?

File: lwgenerate/gcp/gcp.go (hunk 2 above)
Comment (ipcrm) [reply lwgenerate/gcp/gcp.go:538]: Edit: seems like yes; I can't complete the generate flow without a region set

File: cli/cmd/generate_gcp.go [hunk 6]
```diff
@@ -29,6 +30,11 @@ var (
 	QuestionExistingServiceAccountName       = "Specify an existing service account name:"
 	QuestionExistingServiceAccountPrivateKey = "Specify an existing service account private key (base64 encoded):"
 
+	GcpAdvancedOptAgentless      = "Configure additional Agentless options"
+	QuestionGcpProjectFilterList = "Specify a comma seprated list of Google Cloud projects that " +
+		"you want to monitor: (optional)"
+	QuestionGcpRegions = "Specify a comma seprated list of regions to deploy:"
```
Comment (ipcrm) [inline cli/cmd/generate_gcp.go:36]: Is this actually an advanced option? Seems like the generate command fails if this is not populated.

**Review by ipcrm — CHANGES_REQUESTED** [review]
Looking good! I had a few questions.

File: lwgenerate/gcp/gcp.go (hunk 3 above)
Comment (PengyuanZhao) [reply lwgenerate/gcp/gcp.go]: Oh, yes. `[]string` works!

File: lwgenerate/gcp/gcp.go (hunk 5 above)
Comment (PengyuanZhao) [reply lwgenerate/gcp/gcp.go]: Yes, Configuration and Audit Log need to point to a Google Cloud provider by alias when there are multiple ones for Agentless.

Added the condition for `config.Agentless` here.

File: cli/cmd/generate_gcp.go (hunk 6 above)
Comment (PengyuanZhao) [reply cli/cmd/generate_gcp.go]: Good call. Moved this question to the top level so users have to answer it.

**Review by ipcrm — COMMENTED** [review]
Re-tested project level integration and that is working well! 

Looks like integration tests are now failing, however?

**Comment by PengyuanZhao** [comment]
@ipcrm The integration test is fixed in https://github.com/lacework/go-sdk/pull/1506

File: cli/cmd/generate_gcp.go [hunk 7]
```diff
@@ -29,6 +30,11 @@ var (
 	QuestionExistingServiceAccountName       = "Specify an existing service account name:"
 	QuestionExistingServiceAccountPrivateKey = "Specify an existing service account private key (base64 encoded):"
 
+	GcpAdvancedOptAgentless      = "Configure additional Agentless options"
+	QuestionGcpProjectFilterList = "Specify a comma seprated list of Google Cloud projects that " +
```
Comment (afiune) [inline cli/cmd/generate_gcp.go:34]: typo
```suggestion
	QuestionGcpProjectFilterList = "Specify a comma separated list of Google Cloud projects that " +
```

File: cli/cmd/generate_gcp.go [hunk 8]
```diff
@@ -29,6 +30,11 @@ var (
...
+		"you want to monitor: (optional)"
+	QuestionGcpRegions = "Specify a comma seprated list of regions to deploy Agentless:"
```
Comment (afiune) [inline cli/cmd/generate_gcp.go:36]:
```suggestion
	QuestionGcpRegions = "Specify a comma separated list of regions to deploy Agentless:"
```

**Review by afiune — APPROVED** [review]

**Review by ipcrm — APPROVED** [review]

**Pull request merged**